### ISSUE TITLE
Invalid stubs for ExifRead library

### ISSUE THREAD

**Issue by vinicius507** [microsoft/pyright#8496]
**Describe the bug**

Pyright throws misleading diagnostics on the ExitRead library due to pyright internal stubs of the library being incorrect.

**Code or Screenshots**

```python
import exifread  # ExifRead==3.0.0

def parse_exif_data(file: BinaryIO):
    logging.info("Parsing EXIF data from %s", file.name)
    tags = exifread.process_file(file)
    logging.info("Image %s has %d tags", file.name, len(tags))
```

![image](https://github.com/user-attachments/assets/a8d42bda-d0bb-49ee-bdd1-9c4ed080d742)

The `process_file` function has the following signature:

```py
def process_file(fh: BinaryIO, stop_tag=DEFAULT_STOP_TAG,
                 details=True, strict=False, debug=False,
                 truncate_tags=True, auto_seek=True, extract_thumbnail=True)
``` 

While the stubs have the following signature:

```py
def process_file(
    fh: Reader,
    stop_tag: str = "UNDEF",
    details: bool = True,
    strict: bool = False,
    debug: bool = False,
    truncate_tags: bool = True,
    auto_seek: bool = True,
) -> dict[str, Any]: ...
```

**Environment**

- **OS:** `Linux desktop 6.6.37 #1-NixOS SMP PREEMPT_DYNAMIC Fri Jul 5 07:34:07 UTC 2024 x86_64 GNU/Linux`
- **Neovim:** 0.10.0
- **Python:** 3.11.8
- **Pyright:** 1.1.370

**Comment by erictraut** [comment]
Pyright bundles stubs from the community-maintained [typeshed project](https://github.com/python/typeshed/tree/main/stubs/ExifRead). If you think there's a bug in these stubs, please file a bug report or PR in that project.

**Comment by vinicius507** [comment]
Awesome. Thanks @erictraut. You may close the issue if need be.
